packages/core/src/main/conversation/ConversationService/ConversationService.ts

```diff
@@ -1347,21 +1283,26 @@ export class ConversationService {
    * @param epoch The current epoch of the local conversation
    */
   public async sendExternalJoinProposal(conversationGroupId: string, epoch: number) {
-    const coreCryptoClient = this.coreCryptoClientProvider();
     const groupIdDecodedFromBase64 = Decoder.fromBase64(conversationGroupId!).asBytes;
-    const externalProposal = await coreCryptoClient.newExternalProposal(ExternalProposalType.Add, {
+    const externalProposal = await this.mlsService.newExternalProposal(ExternalProposalType.Add, {
       epoch,
       conversationId: groupIdDecodedFromBase64,
     });
     await this.apiClient.api.conversation.postMlsMessage(
       //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
       optionalToUint8Array(externalProposal),
     );
+
+    //We store the info when user was added (and key material was created), so we will know when to renew it
+    const userAddedTimeStamp = new Date().getTime();
+    await this.notificationService.storeLastKeyMaterialUpdateDate({
```

Comment:
@PatrykBuniX  Can we make it a function and pass the timestamp as its repeated above at line number 1179?


await this.notificationService.storeLastKeyMaterialUpdateDate({
      previousUpdateDate: groupCreationTimeStamp,
      groupId,
    });

Reply:
Sure, why not! 😄